app/views/hyrax/base/_currently_shared.html.erb

```diff
@@ -0,0 +1,63 @@
+<%#
+# form object.class = SimpleForm::FormBuilder
+#   For works (i.e. GenericWork):
+#   - form object.object = Hyrax::GenericWorkForm
+#   - form object.object.model = GenericWork
+#   - use the work itself
+#   For file_sets:
+#   - form object.object.class = FileSet
+#   - use work the file_set is in
+#   No other object types are supported by this view. %>
+<% if f.object.respond_to?(:model) && f.object.model.work?
```

Comment:
a good follow up might be to refactor this into a helper? it's only called once, and it's a bit worrying to have this branching logic in a view.

or maybe it's not a helper and could be wrapped into the service object initializer.

Reply:
How about `permission_service = Hyrax::EditPermissionsService.build_service_from(form: f, ability: current_ability)`